core/domain/learner_progress_domain_test.py

```diff
@@ -0,0 +1,79 @@
+# limitations under the License.
+
+"""Testing Domain objects for learner progress."""
+
+from core.domain import learner_progress_domain
+from core.tests import test_utils
+import feconf
+
+
+class LearnerProgressDomainUnitTests(test_utils.GenericTestBase):
+    """Test the Learner Progress domain."""
+
+    def setUp(self):
+        super(LearnerProgressDomainUnitTests, self).setUp()
+
+        self.exploration_keys_lst = []
+        self.collection_keys_lst = []
+        # iterate over DEMO_EXPLORATIONS.
+        # Calculates length in case we have less DEMO members than needed.
+        self.num_of_demo_explorations = len(feconf.DEMO_EXPLORATIONS)
+        self.num_of_demo_collections = len(feconf.DEMO_COLLECTIONS)
+        exploration_keys_iter = feconf.DEMO_EXPLORATIONS.iterkeys()
+        collection_keys_iter = feconf.DEMO_COLLECTIONS.iterkeys()
+        for _ in xrange(0, self.num_of_demo_explorations):
+            self.exploration_keys_lst.append(next(exploration_keys_iter))
+        for _ in xrange(0, self.num_of_demo_collections):
+            self.collection_keys_lst.append(next(collection_keys_iter))
+
+    def test_init_learner_progress(self):
+        learner_progress = learner_progress_domain.LearnerProgress(
+            feconf.DEMO_EXPLORATIONS[self.exploration_keys_lst[0]],
+            feconf.DEMO_COLLECTIONS[self.collection_keys_lst[0]],
+            feconf.DEMO_EXPLORATIONS[
+                self.exploration_keys_lst[1 % self.num_of_demo_explorations]],
+            feconf.DEMO_COLLECTIONS[
+                self.collection_keys_lst[1 % self.num_of_demo_collections]],
+            feconf.DEMO_EXPLORATIONS[
+                self.exploration_keys_lst[2 % self.num_of_demo_explorations]],
+            feconf.DEMO_COLLECTIONS[
+                self.collection_keys_lst[2 % self.num_of_demo_collections]])
+
+        self.assertIsNotNone(learner_progress)
+
+
+class ActivityIdsInLearnerDashboardDomainTests(LearnerProgressDomainUnitTests):
+    """Test the ActivityIdsInLearnerDashboard domain."""
+
+    def setUp(self):
+        super(ActivityIdsInLearnerDashboardDomainTests, self).setUp()
+
+        self.learner = learner_progress_domain.ActivityIdsInLearnerDashboard(
+            feconf.DEMO_EXPLORATIONS[self.exploration_keys_lst[0]],
+            feconf.DEMO_COLLECTIONS[self.collection_keys_lst[0]],
+            feconf.DEMO_EXPLORATIONS[
+                self.exploration_keys_lst[1 % self.num_of_demo_explorations]],
+            feconf.DEMO_COLLECTIONS[
+                self.collection_keys_lst[1 % self.num_of_demo_collections]],
+            feconf.DEMO_EXPLORATIONS[
+                self.exploration_keys_lst[2 % self.num_of_demo_explorations]],
+            feconf.DEMO_COLLECTIONS[
+                self.collection_keys_lst[2 % self.num_of_demo_collections]])
+
+    def test_to_dict(self):
+        learner_dict = self.learner.to_dict()
+        self.assertIsNotNone(learner_dict)
```

Comment:
I think a better way to check this would be to actually compare the contents of the dictionary, learner_dict.